### PR TITLE
[EI-4772] - Add support for batch PATCH and DELETE endpoints

### PR DESCRIPTION
**Pull request by kiyaGu**
## Why?

-  We have added batch `PATCH` and `DELETE` endpoints to the API and the client needs to support that

## What?

- Adds support for batch `PATCH` and `DELETE` endpoints
- Add unit tests

### CODE REVIEW

File: lib/api/batch.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+const urlJoin = require('url-join');
+const queryString = require('../utils/queryString');
+
+/**
+ * @param {string} baseUrl - the base url for node actions. Will default to '/v1/batch' if no `batchEndpoint` option is provided.
+ * @param {import('../BizOpsClient').FactoryOptions} options
+ */
+function buildPatch(baseUrl, options) {
```
Comment (GeoffThorpeFT) [inline lib/api/batch.js:8]: I think this should be named `buildBatch` as it acts as the "parent" for the two methods that are defined below.

File: lib/api/batch.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+		});
+	}
+
+	return {
+		patch: batchPatch,
+		delete: batchDelete,
+	};
+}
+
+module.exports = buildPatch;
```
Comment (GeoffThorpeFT) [inline lib/api/batch.js:55]: I think this should be exporting `buildBatch` as it acts as the "parent" for the two methods that are defined above.

Comment (kiyaGu) [reply lib/api/batch.js:55]: 🙈

**Comment by GeoffThorpeFT** [comment]
I think you may have missed a change. Were you aware that the `BizOpsClient.js` file is the "controller" for all the endpoints? You need to change that file to contain a require and a call to `buildBatch` to trigger the activation of the endpoints defined in `batch.js`

**Comment by kiyaGu** [comment]
> I think you may have missed a change. Were you aware that the `BizOpsClient.js` file is the "controller" for all the endpoints? You need to change that file to contain a require and a call to `buildBatch` to trigger the activation of the endpoints defined in `batch.js`

@GeoffThorpeFT thanks for pointing that out. I have done that now.

**Comment by GeoffThorpeFT** [comment]
Nice. I've just remembered that the README also needs to be updated as it lists all the available methods and we haven't mentioned `batch` 😄

**Review by GeoffThorpeFT — APPROVED** [review]
Many thanks 😄

**Pull request merged**